app/helpers/cms/form_helper.rb

```diff
@@ -1,59 +1,29 @@
+# frozen_string_literal: true
+
 module CMS
   module FormHelper
 
     def cms_form_for(template, options = {}, &block)
-      base_name = template.class.base_class.model_name.to_s.parameterize(separator: '_').to_sym
-
       options[:builder] ||= CMS::SemanticFormBuilder
       options[:url] ||= polymorphic_path([:provider, :admin, template])
-      options[:as] ||= base_name
+      options[:as] ||= base_name(template)
 
       semantic_form_for(template, options, &block)
     end
 
     def cms_remote_form_for(template, options = {}, &block)
-      cms_form_for(template, options.reverse_merge(:html => { :"data-remote" => 'true' }), &block)
+      cms_form_for(template, options.reverse_merge({ remote: true }), &block)
     end
 
     def cms_fields_for(template, options = {}, &block)
-      base_name = template.class.base_class.model_name.to_s.parameterize(separator: '_').to_sym
-
       options[:builder] ||= CMS::SemanticFormBuilder
-      options[:as] ||= base_name
```

Comment:
Unlike `form_for`, I see that `fields_for` doesn't take into account the `options[:as]`

Reply:
this is same what is changed here just removed monkey patching right?

Reply:
if we have not done the above change then also it will be same

Reply:
I'm not sure, what you mean by "this is same"? Same as what?

This indeed does not contribute to fixing the original issue, this is just refactoring, because this monkey-patch is not needed anymore?

Reply:
I agree with removing monkey patch , also that issue of removing error class from input class
but what fixes that prams issue?

or its like removing monkey patch fixes issue , because we upgraded Formtastic??

Reply:
see https://github.com/3scale/porta/pull/3656/files#r1431199745